File: peek_plugin_docdb/_private/tuples/DocumentUpdateDateTuple.py
from typing import Dict

from peek_plugin_docdb._private.PluginNames import docDbTuplePrefix
from vortex.Tuple import addTupleType, TupleField, Tuple


@addTupleType
class DocumentUpdateDateTuple(Tuple):
    """ DocDb Object Update Date Tuple

    This tuple represents the state of the chunks in the cache.
    Each chunkKey has a lastUpdateDate as a string, this is used for offline caching
    all the chunks.
    """
    __tupleType__ = docDbTuplePrefix + "DocumentUpdateDateTuple"

    initialLoadComplete: bool = TupleField()
    updateDateByChunkKey: Dict[str, str] = TupleField({})

File: peek_plugin_docdb/plugin-module/_private/document-loader/DocumentUpdateDateTuple.ts
import {addTupleType, Tuple} from "@synerty/vortexjs";
import {docDbTuplePrefix} from "../PluginNames";


@addTupleType
export class DocumentUpdateDateTuple extends Tuple {
    public static readonly tupleName = docDbTuplePrefix + "DocumentUpdateDateTuple";

    initialLoadComplete: boolean = false;
    updateDateByChunkKey: {} = {};

    constructor() {
        super(DocumentUpdateDateTuple.tupleName)
    }
}